Making Makefile variables accessible from from make task.

robert-zaremba opened the pull request:
When I tried to install Pymacs on my Arch linux, i fall into problems, since Arch linux has default python as a python3, and when i set PYTHON variable in Makefile it wasn't recognized in the scirpts.
So the commit is about:

Made visable environment variables: PYTHON, EMACS, for make task based
on values in Makefile header.
This is usefull take default values based on the ones in
Makefile. Previously if we modyfied Makefile PYTHON value the check task
perform check agains PYTHON vale, but prepare didn't.

Commit:
Setting environment variables for make task

Made visable environment variables: PYTHON, EMACS, for make task based
on values in Makefile header.
This is usefull take default values based on the ones in
Makefile. Previously if we modyfied Makefile PYTHON value the check task
perform check agains PYTHON vale, but prepare didn't.

pinard commented:
Hello, Robert.

I see your diffs, but do not understand the problem they intend to solve.   PPPP is already defined as "$(PYTHON) pppp" (so it is already called with the Python defined by PYTHON), and pppp itself does not depend at all on environment variables EMACS or PYTHON.  So, setting EMACS or PYTHON prior to calling pppp should not have any effect.

You should normally be able to use "make PYTHON=some-python" without even having to modify the Makefile.  I presume this is explained in the documentation.  Am I missing something?  Should the documentation be clarified?

robert-zaremba commented:
That's right. "make PYTHON=some-python" works great - so you specify python bin through env setting.
But this is inconsistent with make check task, as it set the python version defined in Makefile.

So 'make check PYTHON=some-python" and 'make PYTHON=some-python' set different version of python in configuration files. Look how PYTHON variable is defined in ppppconfig.py file (line 36) and add some debug log/print and run it with both upper commands. 

This is just a comment, if you consider this change is worthless just reject :)

Commit:
Merge remote-tracking branch 'pinard/master'

Commit:
chagene pymacs-timeout-at-start

In the new version of Emacs (and probably python-mode) this value makes
error: *Pymacs helper did not start within 30 seconds*
Incrementing it prohibits this error and makes loading pymacs seamless

robert-zaremba commented:
Hello. I've dig this PR out. Is it possible to merge it?

punchagan commented:
@robert-zaremba https://github.com/pinard/Pymacs/issues/34#issuecomment-222578942
